streams/src/main/java/org/apache/kafka/streams/state/internals/ThreadCache.java

```diff
@@ -201,6 +201,14 @@ public MemoryLRUCacheBytesIterator reverseAll(final String namespace) {
         return new MemoryLRUCacheBytesIterator(cache.reverseAllKeys(), cache);
     }
 
+    public MemoryLRUCacheBytesIterator prefixScan(final String namespace, final Bytes from, final Bytes to) {
+        final NamedCache cache = getCache(namespace);
+        if (cache == null) {
+            return new MemoryLRUCacheBytesIterator(Collections.emptyIterator(), new NamedCache(namespace, this.metrics));
+        }
+        return new MemoryLRUCacheBytesIterator(cache.keyRange(from, to, false), cache);
```

Comment:
If you consider my comment in `NamedCache`, you could also just call `range()` here. Or -- if you want to be more descriptive -- having `prefixScan()` and `range()` in this class calling a private overload of `range()` with a flag that excludes or includes the end of the range. That would deduplicate code.